clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -50,15 +52,20 @@ private ClientUtils() {}
                     if (host == null || port == null)
                         throw new ConfigException("Invalid url in " + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG + ": " + url);
 
-                    InetSocketAddress address = new InetSocketAddress(host, port);
-
-                    if (address.isUnresolved()) {
-                        log.warn("Removing server {} from {} as DNS resolution failed for {}", url, CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, host);
-                    } else {
-                        addresses.add(address);
-                    }
+                        InetAddress[] inetAddresses = InetAddress.getAllByName(host);
```

Comment:
is this part of the fix for brokers switching clusters, or a backport of another fix? if its a backport, do we want to make it a different PR ?

Reply:
this is a backport of 2.4 client fix, sure, will open another PR for this.